BoneReconstructionPlanner/BoneReconstructionPlanner.py

```diff
@@ -793,23 +796,33 @@ def generateFibulaPlanes(self):
       modelNodeItemID = shNode.GetItemByDataNode(modelNode)
       shNode.SetItemParent(modelNodeItemID, cutBonesFolder)
   
-  def getRotationMatrixFromAxis1ToAxis2(self,axis1,axis2):
+  def getAxes1ToWorldRotationMatrix(self,axis1X,axis1Y,axis1Z):
     rotationMatrix = vtk.vtkMatrix4x4()
     rotationMatrix.DeepCopy((1, 0, 0, 0,
                              0, 1, 0, 0,
                              0, 0, 1, 0,
                              0, 0, 0, 1))
-    #range(3) because rotation matrix is 3x3, the forth colomn and forth row are unchanged
-    for i in range(3):
-      for j in range(3):
-        projection = [0,0,0]
-        vtk.vtkMath.ProjectVector(axis2[i],axis1[j],projection)
-        if vtk.vtkMath.Dot(projection,axis1[j]) >= 0:
-          rotationMatrix.SetElement(j,i,vtk.vtkMath.Norm(projection))
-        else:
-          rotationMatrix.SetElement(j,i,-vtk.vtkMath.Norm(projection))
-    
+    rotationMatrix.SetElement(0,0,axis1X[0])
+    rotationMatrix.SetElement(0,1,axis1X[1])
+    rotationMatrix.SetElement(0,2,axis1X[2])
+    rotationMatrix.SetElement(1,0,axis1Y[0])
+    rotationMatrix.SetElement(1,1,axis1Y[1])
+    rotationMatrix.SetElement(1,2,axis1Y[2])
+    rotationMatrix.SetElement(2,0,axis1Z[0])
+    rotationMatrix.SetElement(2,1,axis1Z[1])
+    rotationMatrix.SetElement(2,2,axis1Z[2])
+
     return rotationMatrix

```

Comment:
This code works well. But in my opinion it shouldn't because of the mathematical definition of Axes1ToWorldRotationMatrix the indexes in the SetElement function should be swapped
https://en.wikibooks.org/wiki/Robotics_Kinematics_and_Dynamics/Description_of_Position_and_Orientation

Reply:
It is correct to store vectors in columns or rows. But then you need to multiply from left/right accordingly. In practice, in a library you choose one or the other convention and use it consistently.

In medical image computing, most commonly axis vectors are in columns and vectors are multiplied from the left (A*v). We use this convention, too, so the matrix setting is now incorrect. It works correctly now, because there is another mistake somewhere and the two cancel each other out. 

Most likely you missed a matrix inversion somewhere (since transpose(A) =inv(A) for orthonormal matrices), which is expected, if you don't name variables correctly. For example, you should never write down something like "rotationMatrix", ever. You must always include name of the "from" and "to" coordinate systems. For example, rotationAtoB.

Reply:
I checked all intervening functions definitions and calls and I couldn't find anything wrong. Could you try finding the mistake?

Reply:
I'll try rewriting all the code that doesn't work from zero. Do you have any code examples of Axes1ToAxes2 transforms?

Reply:
I created a small test script to check what was working right and what wasn't (it is published). I suspected that the error was inside _getAxes1ToWorldRotationMatrix_ because I had tested _getAxes1ToAxes2RotationMatrix_ with two arbitrary matrices and the result was mathematically correct. Surprisingly, using the native planeToWorldTransform function to create the plane0ToPlane1RotationMatrix makes the transform incorrect and the transform is only correct when you create it using the results from (the wrongful) definition of _getAxes1ToWorldRotationMatrix_. This suggests me that this behaviour is due to a bug in Slicer code.

Reply:
By `planeToWorldTransform` which class::method do you mean exactly?

Reply:
I meant GetPlaneToWorldMatrix

Reply:
Andras did I explained myself in a clear enough way? Were you able to reproduce the bug using the script I uploaded?

Reply:
I'll check it this afternoon.

Reply:
Hi Andras. Did I explained myself in a clear enough way? Were you able to reproduce the bug using the script I uploaded?

Reply:
Hi Andras. How are you? 
Do I go on with the merge anyway?

Reply:
I'll try to review this today (Friday) if you don't hear from me by the end of the day then you can merge it.